api/libs/puvspr-calculations/src/puvspr-calculations.service.ts

```diff
@@ -0,0 +1,143 @@
+import { ScenarioFeaturesData } from '@marxan/features';
+import { Inject, Injectable } from '@nestjs/common';
+import { EntityManager } from 'typeorm';
+
+export const geoEntityManagerToken = Symbol('geo entity manager token');
+
+@Injectable()
+export class PuvsprCalculationsService {
+  constructor(
+    @Inject(geoEntityManagerToken)
+    private readonly geoEntityManager: EntityManager,
+  ) {}
+
+  public async computeSpeciesId(
+    featureIds: string[],
+    scenarioId: string,
+  ): Promise<
+    {
+      specieId: number;
+      featureId: string;
+    }[]
+  > {
+    return this.geoEntityManager
+      .createQueryBuilder()
+      .select('min(sfd.feature_id)', 'specieId')
+      .addSelect('sfd.api_feature_id', 'featureId')
+      .from(ScenarioFeaturesData, 'sfd')
+      .where('sfd.api_feature_id IN (:...featureIds)', {
+        featureIds,
+      })
+      .andWhere('sfd.scenario_id = :scenarioId', { scenarioId })
+      .groupBy('sfd.api_feature_id')
+      .execute();
+  }
+
+  public async computeMarxanAmountPerPlanningUnit(
+    featureId: string,
+    scenarioId: string,
+  ) {
+    /**
+     * @TODO further performance savings: limiting scans to planning_units_geom
+     * by partition (we need to get the grid shape from the parent project); use
+     * && operator instead of st_intersects() for bbox-based calculation of
+     * intersections.
+     *
+     * @TODO Calculate `amount` correctly from `amount_from_legacy_project`. The
+     * initial implementation is simply a placeholder so that we can use
+     * `amount_from_legacy_projects` in legacy project piece importers.
```

Comment:
this can be removed as you've basically taken care of it via the `computeLegacyAmountPerPlanningUnit` method below